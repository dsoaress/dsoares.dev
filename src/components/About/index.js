import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'

import BlockContent from '../BlockContent'
import Social from '../Social'
import * as S from './styled'

const About = () => {
  const { sanitySiteSettings } = useStaticQuery(
    graphql`
      query {
        sanitySiteSettings {
          bio {
            title
            image {
              asset {
                fluid(maxWidth: 420) {
                  ...GatsbySanityImageFluid
                }
              }
            }
            _rawText
          }
        }
      }
    `
  )

  return (
    <S.Wrapper>
      <h1>{sanitySiteSettings.bio.title}</h1>
      <S.Grid>
        <S.TextWrapper>
          {sanitySiteSettings.bio._rawText && (
            <BlockContent blocks={sanitySiteSettings.bio._rawText} />
          )}
          <Social />
        </S.TextWrapper>
        <S.Image fluid={sanitySiteSettings.bio.image.asset.fluid} />
      </S.Grid>
    </S.Wrapper>
  )
}

export default About
