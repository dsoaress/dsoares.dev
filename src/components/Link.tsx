import cn from 'classnames'
import NextLink from 'next/link'
import type { AnchorHTMLAttributes, ReactNode } from 'react'
import { Link as ReactScroll } from 'react-scroll'

type LinkProps = {
  active?: boolean
  href: string
  locale?: string
  external?: boolean
} & AnchorHTMLAttributes<HTMLAnchorElement>

type ScrollLinkProps = {
  to: string
  className?: string
  children: ReactNode
}

export function Link({ active = false, href, className, locale, external, ...props }: LinkProps) {
  return (
    <NextLink href={href} locale={locale}>
      <a
        className={cn(className, { 'text-primary-500 dark:text-primary-500': active })}
        rel={external ? 'noopener noreferrer' : undefined}
        target={external ? '_blank' : undefined}
        {...props}
      />
    </NextLink>
  )
}

export function ScrollLink(props: ScrollLinkProps) {
  // TODO fix this
  // @ts-ignore
  return <ReactScroll smooth={true} offset={-76} href="/" aria-hidden {...props} />
}

export function CardLink({ className, ...props }: LinkProps) {
  return <Link className={cn('absolute inset-0', className)} {...props} />
}
