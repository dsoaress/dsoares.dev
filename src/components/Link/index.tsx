import NextLink from 'next/link'
import { AnchorHTMLAttributes } from 'react'

import { StyledLink } from './styles'

export { CardLink, ScrollLink } from './styles'

export type LinkProps = {
  active?: boolean
  href: string
  locale?: string
} & AnchorHTMLAttributes<HTMLAnchorElement>

export function Link({ active = false, href, locale, ...rest }: LinkProps) {
  return (
    <NextLink href={href} locale={locale} passHref>
      <StyledLink active={active} {...rest} />
    </NextLink>
  )
}
