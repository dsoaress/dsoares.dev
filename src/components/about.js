import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

import BlockContent from './block-content'
import Social from './social'

const About = () => {
  const { sanitySiteSettings } = useStaticQuery(
    graphql`
      query {
        sanitySiteSettings {
          bio {
            title
            image {
              asset {
                fluid(maxWidth: 420) {
                  ...GatsbySanityImageFluid
                }
              }
            }
            _rawText
          }
        }
      }
    `
  )

  return (
    <div>
      <h1>{sanitySiteSettings.bio.title}</h1>
      <div>
        <div>
          {sanitySiteSettings.bio._rawText && (
            <BlockContent blocks={sanitySiteSettings.bio._rawText} />
          )}
          <Social />
        </div>
        <Img fluid={sanitySiteSettings.bio.image?.asset.fluid} />
      </div>
    </div>
  )
}

export default About
