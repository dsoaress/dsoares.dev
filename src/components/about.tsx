import { Block } from './block'

export function About() {
  return (
    <Block className="col-span-12 text-3xl leading-snug">
      <p className="text-neutral-400">
        Specializing in <span className="text-neutral-50">scalable, robust applications</span>, I
        focus on delivering seamless user experiences and optimizing backend systems for maximum
        efficiency.
      </p>
    </Block>
  )
}
