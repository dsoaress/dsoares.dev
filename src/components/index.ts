export * from './about'
export * from './block'
export * from './footer'
export * from './header'
export * from './home-container'
export * from './logo'
export * from './social'
