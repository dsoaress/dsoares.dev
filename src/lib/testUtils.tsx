import { render } from '@testing-library/react'
import mockRouter from 'next-router-mock'
import { ReactElement, ReactNode } from 'react'

import { i18n } from '@/localesConfig'
import { Providers } from '@/providers'

type ReactProvidersProps = {
  children?: ReactNode
}

jest.mock('next/router', () => require('next-router-mock'))

const ReactProviders = ({ children }: ReactProvidersProps) => {
  return <Providers>{children}</Providers>
}

const customRender = (ui: ReactElement, options = {}) => {
  mockRouter.locales = i18n.locales
  mockRouter.locale = i18n.defaultLocale
  render(ui, { wrapper: ReactProviders, ...options })
}
export * from '@testing-library/react'

export { customRender as render }
