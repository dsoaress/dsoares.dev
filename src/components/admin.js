import { useState } from 'react'
import useSWR, { mutate } from 'swr'
import { signOut } from 'next-auth/client'

import Spinner from '@/components/spinner'
import { fetcher } from '@/lib/fetcher'

export default function Tasks() {
  const [label, setLabel] = useState('')
  const [url, setUrl] = useState('')

  const { data } = useSWR('/api', fetcher, {
    refreshInterval: 500
  })

  if (!data) return <Spinner />

  async function handleSubmit(e) {
    e.preventDefault()
    mutate('/api', { ...data, data: { links: { label, url } } }, false)

    await fetch('/api', {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ link: { label, url } })
    })

    mutate('/api')
    setLabel('')
    setUrl('')
  }

  return (
    <div>
      <div>
        <button onClick={() => signOut()}>Sign out</button>
      </div>
      <form onSubmit={handleSubmit}>
        <div className="text-black">
          <label>Label</label>
          <input
            type="text"
            value={label}
            onChange={e => setLabel(e.target.value)}
            required
          />
          <label>Url</label>
          <input
            type="text"
            value={url}
            onChange={e => setUrl(e.target.value)}
            required
          />
        </div>
        <button type="submit">Add</button>
      </form>
      <div>
        {!data.links || data.links.length === 0 ? (
          <p>
            It looks like you don't have any more tasks... Would you like to add
            a new one?
          </p>
        ) : (
          data.links.map(link => (
            <div key={link._id}>
              <p>{link.label}</p>
              <p>{link.url}</p>
            </div>
          ))
        )}
      </div>
    </div>
  )
}
