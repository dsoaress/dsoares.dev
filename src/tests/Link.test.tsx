import { Link } from '@/components/Link'
import { render, screen } from '@/lib/testUtils'

describe('<Link />', () => {
  it('Renders with children', () => {
    render(<Link href="#">Children</Link>)

    expect(screen.getByText('Children')).toBeInTheDocument()
    expect(screen.getByText('Children')).toMatchSnapshot()
  })

  it('Renders with active className', () => {
    render(
      <Link href="#" active>
        Children
      </Link>
    )

    expect(screen.getByText('Children')).toMatchSnapshot()
  })
})
