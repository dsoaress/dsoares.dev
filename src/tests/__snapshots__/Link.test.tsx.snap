// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Renders with active className 1`] = `
.c0 {
  z-index: 10;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition-property: background-color,border-color,color,fill,stroke;
  transition-property: background-color,border-color,color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  cursor: pointer;
  --tw-text-opacity: 1;
  color: rgba(244,63,94,var(--tw-text-opacity));
}

.c0:hover {
  --tw-text-opacity: 1;
  color: rgba(244,63,94,var(--tw-text-opacity));
}

.c0:active {
  --tw-text-opacity: 1;
  color: rgba(225,29,72,var(--tw-text-opacity));
}

<a
  class="c0"
  href="/#"
>
  Children
</a>
`;

exports[`<Link /> Renders with children 1`] = `
.c0 {
  z-index: 10;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition-property: background-color,border-color,color,fill,stroke;
  transition-property: background-color,border-color,color,fill,stroke;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  cursor: pointer;
}

.c0:hover {
  --tw-text-opacity: 1;
  color: rgba(244,63,94,var(--tw-text-opacity));
}

.c0:active {
  --tw-text-opacity: 1;
  color: rgba(225,29,72,var(--tw-text-opacity));
}

<a
  class="c0"
  href="/#"
>
  Children
</a>
`;
